### PR TITLE
Updates credential details to support plugin and prompt fields

### PR DESCRIPTION
**Pull request by mabashian**
##### SUMMARY
Link #7127 

This PR accomplishes two things:

1) Support display of prompt on launch password fields in the Details view:
Screen Shot 2020-06-08 at 11 00 58 AM

- Previously we were showing the `ASK` text that the API returns.  I think this is a little bit more informative and it matches the pattern that we have on the JT details view.

2) Support display of plugin configured fields in the Details view:
Screen Shot 2020-06-08 at 11 01 29 AM

- Since we added support for credential plugins with https://github.com/ansible/awx/pull/7125 we needed to update the details view to display that plugin configuration.  Configuration consists of two things: a credential and some metadata.  After talking with @trahman73 we decided that the path of least resistance was to display the metadata as JSON in a codemirror underneath the field.  This way, a user can see both the credential and the metadata used when pulling this field from an external source.

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
 - UI

### CODE REVIEW

File: awx/ui_next/src/screens/Credential/CredentialDetail/CredentialDetail.jsx
Comment (jlmitch5) [inline awx/ui_next/src/screens/Credential/CredentialDetail/CredentialDetail.jsx:26]: aren't codemirrors already full-length?

File: awx/ui_next/src/screens/Credential/CredentialDetail/CredentialDetail.jsx
Comment (jlmitch5) [inline awx/ui_next/src/screens/Credential/CredentialDetail/CredentialDetail.jsx]: since we keep these functions in our api file now, we could do the recursive to last page thing there...would be a bit more elegant than keeping it in the views like we had to do in the old app.  but i'm also fine for keeping things simple like this.

**Review by jlmitch5 — COMMENTED** [review]
looks good, just a couple questions

File: awx/ui_next/src/screens/Credential/CredentialDetail/CredentialDetail.jsx
Comment (mabashian) [reply awx/ui_next/src/screens/Credential/CredentialDetail/CredentialDetail.jsx:26]: If we don't make it take up all the columns of the grid then it ends up looking like:

Screen Shot 2020-06-09 at 4 55 13 PM

File: awx/ui_next/src/screens/Credential/CredentialDetail/CredentialDetail.jsx
Comment (mabashian) [reply awx/ui_next/src/screens/Credential/CredentialDetail/CredentialDetail.jsx]: @jlmitch5 see if https://github.com/ansible/awx/pull/7281/commits/834cfb82f8da49003633743599640d405c92cf47 is what you're looking for.  I like this pattern for requests that we know will be made recursively.

Comment (jakemcdermott) [reply awx/ui_next/src/screens/Credential/CredentialDetail/CredentialDetail.jsx]: I'm ok with that approach if the recursion is capped to something reasonable and is guaranteed to terminate regardless of whether the server sends us correct data. A max request limit and counter will suffice.

File: awx/ui_next/src/screens/Credential/CredentialDetail/CredentialDetail.jsx
Comment (mabashian) [inline awx/ui_next/src/screens/Credential/CredentialDetail/CredentialDetail.jsx:51]: Reminder: useRequest

File: awx/ui_next/src/screens/Credential/CredentialDetail/CredentialDetail.jsx
Comment (mabashian) [reply awx/ui_next/src/screens/Credential/CredentialDetail/CredentialDetail.jsx]: Sounds good @jakemcdermott - I added that logic and capped it at 5 requests.  The likelihood of more than 1000 input sources on a cred seems very low.

Comment (mabashian) [reply awx/ui_next/src/screens/Credential/CredentialDetail/CredentialDetail.jsx]: We may decide that we want to build some kind of generic wrapper around this sort of logic so that we can recursively fetch from other endpoints without having to duplicate a lot of this.

File: awx/ui_next/src/screens/Credential/CredentialDetail/CredentialDetail.jsx
Comment (mabashian) [reply awx/ui_next/src/screens/Credential/CredentialDetail/CredentialDetail.jsx:51]: @marshmalien - added this in.  Thanks for the snippet!

File: awx/ui_next/src/api/models/Credentials.js
Comment (AlexSCorey) [inline awx/ui_next/src/api/models/Credentials.js:24]: At first this function was a bit confusing, but after looking at it again I like it.

File: awx/ui_next/src/screens/Credential/CredentialDetail/CredentialDetail.jsx
Comment (AlexSCorey) [inline awx/ui_next/src/screens/Credential/CredentialDetail/CredentialDetail.jsx]: Why use `<Trans>` rather than doing 
```{i18n._(t * This field will be retrieved from an external secret management system using the specified credential.)```
? which has been our pattern to date

**Review by AlexSCorey — COMMENTED** [review]
I think this looks good.  Just 1 thing in question about `<Trans>`.

File: awx/ui_next/src/screens/Credential/CredentialDetail/CredentialDetail.jsx
Comment (mabashian) [reply awx/ui_next/src/screens/Credential/CredentialDetail/CredentialDetail.jsx]: Oh I thought we had decided to use Trans when the string stands alone in the DOM (ie not in an attribute and doesn't live outside the render function).  I'll change this.  There's no real benefit to one over the other in this case I don't think.

File: awx/ui_next/src/api/models/Credentials.js
Comment (jakemcdermott) [inline awx/ui_next/src/api/models/Credentials.js]: The credential edit component expects this return structure to be `{ data: { results: [...] } }`.

**Review by jakemcdermott — CHANGES_REQUESTED** [review]

File: awx/ui_next/src/api/models/Credentials.js
Comment (jakemcdermott) [inline awx/ui_next/src/api/models/Credentials.js]:
```suggestion
        if (data?.next && requestCounter <= maxRequests) {
```

File: awx/ui_next/src/api/models/Credentials.js
Comment (jakemcdermott) [inline awx/ui_next/src/api/models/Credentials.js:50]: Need to surface this error in [CredentialEdit.jsx#L154](https://github.com/ansible/awx/blob/devel/awx/ui_next/src/screens/Credential/CredentialEdit/CredentialEdit.jsx#L154)

```jsx
return <ContentError error={error} />;
```

File: awx/ui_next/src/screens/Credential/CredentialDetail/CredentialDetail.jsx
Comment (jakemcdermott) [inline awx/ui_next/src/screens/Credential/CredentialDetail/CredentialDetail.jsx:102]: No need to nest this function definition inside of the `CredentialDetail`, it can be moved up to module scope and out of the detail component.

File: awx/ui_next/src/api/models/Credentials.js
Comment (mabashian) [reply awx/ui_next/src/api/models/Credentials.js]: https://github.com/ansible/awx/pull/7281/commits/bbafeadd0feeecf393b9ccc719485d01d242ef9d

File: awx/ui_next/src/api/models/Credentials.js
Comment (mabashian) [reply awx/ui_next/src/api/models/Credentials.js]: https://github.com/ansible/awx/pull/7281/commits/b1a5b7a76750cce4538ca57f5c4d832264604b82

File: awx/ui_next/src/api/models/Credentials.js
Comment (mabashian) [reply awx/ui_next/src/api/models/Credentials.js:50]: https://github.com/ansible/awx/pull/7281/commits/abb7486eb06399a7ce6facd516925118d6e02a30

**Review by jakemcdermott — APPROVED** [review]

**Comment by unlikelyzero** [comment]
This PR is blocked by https://github.com/ansible/awx/issues/7814

**Comment by mabashian** [comment]
OK @unlikelyzero I rebased after the PR for #7814 merged so I think this should be unblocked

**Comment by unlikelyzero** [comment]
Awesome. I'm heads down on fixing the credential test so I won't be able to get to this today. I'll take it up first thing Monday

**Comment by dsesami** [comment]
Having some trouble with prompt on launch.

1. Make a machine credential. Take all of the possible fields and set ALL of them to prompt on launch.
2. Edit a basic job template to use this machine credential.
3. Launch the job.

The prompt on launch modal does not prompt for the required fields -- it simply shows the final preview. Attempting to launch results in a 400.
Screenshot from 2020-09-11 13-22-39

**Comment by dsesami** [comment]
This is going to be rebased when https://github.com/ansible/awx/issues/7015 is complete. After that, it can be put in needs_test for final review.

**Comment by unlikelyzero** [comment]
Covered by 5627

**Pull request merged**
